integrationtest/TargetProjects/TargetProject/String/Utf8StringMagic.cs

```diff
@@ -0,0 +1,32 @@
+using System;
+using System.Linq;
+
+namespace ExampleProject.String
+{
+    public class Utf8StringMagic
+    {
+        public ReadOnlySpan<byte> HelloWorld()
+        {
+            return "Hello"u8 + " "u8 + "World!"u8;
```

Comment:
This should be fixed in this PR, not as a separate defect.

Reply:
I've investigated why this test is failing. It fails during mutated code compilation. Stryker by default add ternary expression to each operand:

So this statement
```
return "Hello"u8 + " "u8 + "World!"u8;
```

is transformed into this:
```
return (StrykerqBEu3bP1rxHLxTW.MutantControl.IsActive(99)?""u8 :"Hello"u8 )+ (StrykerqBEu3bP1rxHLxTW.MutantControl.IsActive(100)?""u8 :" "u8 )+ (StrykerqBEu3bP1rxHLxTW.MutantControl.IsActive(101)?""u8:"World!"u8);
```

It boxes each UTF-8 value to `ReadOnlySpan<byte>()` type. The problem lies with C# and how it allows to concatenate UTF-8 string literals. It allows to concatenate UTF-8 string directly: `"Hello"u8 + " "u8 + "World!"u8`, but not when they are represented as `ReadOnlySpan<byte>`: ~~`new ReadOnlySpan<byte>() + new ReadOnlySpan<byte>()`~~.

Hence, mutated code fails to compile with:
```
CS9047	Operator '+' cannot be applied to operands of type 'ReadOnlySpan<byte>' and 'ReadOnlySpan<byte>' that are not UTF-8 byte representations
```

Do you think it's worth fixing this in this PR as it touches core `MutantControl` injection logic? If so, any new ideas are welcome here. So far, I've only thought of converting UTF-8 strings into arrays and concatenating them manually using LINQ when ternary conditions are used:

```
public ReadOnlySpan<byte> HelloWorld()
{
    return (true ? ""u8 : "Hello"u8).ToArray()
        .Concat((true ? ""u8 : " "u8).ToArray()
            .Concat((true ? ""u8 : "World!"u8).ToArray()))
        .ToArray();
}
```

Reply:
Interesting. In that case, we could maybe include some extra code in the compilation that contains an operator overload for `+` on `ReadOnlySpan<byte>`. During the compilation we already pass some custom code to be compiled together with the source project, so that should be fairly easy to do.

> Do you think it's worth fixing this bug in current PR?

Yes, as this seems like an easy fix we should add to this PR. Otherwise, we risk that this will never be fixed once this PR has been merged.

Reply:
I am not sure adding this operator is needed. 
1. `"Hello"u8 + " "u8 + "World!"u8;` is actually a compile time constant so the compiler concatenate the strings and does not compile any expression. See the actual result:
`    internal static readonly __StaticArrayInitTypeSize=13 5A09E8FA9C77807B24E99C9CF999DEBFAD8441E269EB960E201F61FC3DE20D5A/* Not supported: data(48 65 6C 6C 6F 20 57 6F 72 6C 64 21 00) */;
`
An array of 13 bytes. This is seen as a single string here. As such, it does not really make sense to mutate sub part.
2. In any case, there is no interest in mutating this expression more than once. It is almost certain that if removing `Hello` is killed by a failed test, that very same test will kill removing the space or `world!`. So there is no compelling reason to keep each sub mutation. Note that this remarks is also valid for string concatenation
3. As such, it would make more sense to mutate the whole expression once, which would remove any problems with the missing operator.
4. Adding this operator requires injecting a dedicated `using` statement at the start of any files needing it; meaning it would have to be added everywhere. Furthermore, it could result in compilation errors `ambiguous call` that will require a new roll back logic (to detect it and remove the unnecessary `using` statement.
5. The only consequence of not adding this operand is simply that concatenated u8 strings will not be mutated; which looks alright to me.

TLDR;
I recommend: doing nothing for now and contemplate detecting concatenated constant strings (u8 or otherwise) to be mutated at once. But it would require specific logic within the `ExpressionOrchestrator`